### ISSUE TITLE
Thing: provide more information

### ISSUE THREAD

**Issue by UrsusS** [spacemanspiff2007/HABApp#348]
I'm checking my network and zwave devices regularly for some health conditions.
If these conditions are not met, I inform the user about e.g. the detailed device status and the location of the device.

Unfortunately the the thing item is not providing all information, which is available via the REST-interface (defined in OpenhabThingDefinition). 
I'm missing: 

- location, to give the user a hint where to look for the device
- status.description, to give additional information, why the systems thinks there is an issue

In older OH versions was no usefull information in the status description, but that has changed (with 3.3 ?)
Here some examples of these status descriptions I've collected:
- "description": "Initialisierung oder Gerät im Schlafmodus."
- "description": "slave missing"
- "description": "required properties missing"
- "description": "Node initialising: REQUEST_NIF"
- "description": "Node initialising: PING"
- "description": "Node is not communicating with controller"
- "description": "java.util.concurrent.ExecutionException: java.net.NoRouteToHostException: Keine Route zum Zielrechner"
- "description": "Warten bis 19:08:45 für einen automatischen Wiederverbindungsversuch"

Also the data available via the Thing item seems to be static, while the information available via the REST-interface contains also information which changes frequently. An example is "zwave_lastwakeup", which I use for monitoring the health state of the zwave devices.

**Comment by spacemanspiff2007** [comment]
> Also the data available via the Thing item seems to be static

You can already make HABApp update the thing through the [topic filter](https://habapp.readthedocs.io/en/latest/configuration.html#HABApp.config.models.openhab.Connection.topic_filter) in the `config.yml`.
Just set it to `openhab/things/*`instead of `openhab/things/*/added,openhab/things/*/removed,openhab/things/*/status,openhab/things/*/statuschanged`

> Here some examples of these status descriptions I've collected:

Could you post a a RestAPI response which shows one of those?

**Comment by UrsusS** [comment]
I have a new issue which could be related to the changes of the topic filter.

```
2022-12-30 18:30:47.753 [ERROR] [HABApp.openhab.connection] - SSE request Error: Chunk too big
2022-12-30 18:30:47.757 [ERROR] [HABApp.openhab.connection] - Traceback (most recent call last):
2022-12-30 18:30:47.760 [ERROR] [HABApp.openhab.connection] -   File "/opt/habapp/lib/python3.9/site-packages/HABApp/openhab/connection_handler/http_connection.py", line 228, in start_sse_event_listener
2022-12-30 18:30:47.763 [ERROR] [HABApp.openhab.connection] -     async for event in event_source:
2022-12-30 18:30:47.764 [ERROR] [HABApp.openhab.connection] -   File "/opt/habapp/lib/python3.9/site-packages/aiohttp_sse_client/client.py", line 157, in __anext__
2022-12-30 18:30:47.766 [ERROR] [HABApp.openhab.connection] -     async for line_in_bytes in self._response.content:
2022-12-30 18:30:47.767 [ERROR] [HABApp.openhab.connection] -   File "/opt/habapp/lib/python3.9/site-packages/aiohttp/streams.py", line 35, in __anext__
2022-12-30 18:30:47.767 [ERROR] [HABApp.openhab.connection] -     rv = await self.read_func()
2022-12-30 18:30:47.768 [ERROR] [HABApp.openhab.connection] -   File "/opt/habapp/lib/python3.9/site-packages/aiohttp/streams.py", line 311, in readline
2022-12-30 18:30:47.769 [ERROR] [HABApp.openhab.connection] -     return await self.readuntil()
2022-12-30 18:30:47.770 [ERROR] [HABApp.openhab.connection] -   File "/opt/habapp/lib/python3.9/site-packages/aiohttp/streams.py", line 337, in readuntil
2022-12-30 18:30:47.771 [ERROR] [HABApp.openhab.connection] -     raise ValueError("Chunk too big")
2022-12-30 18:30:47.772 [ERROR] [HABApp.openhab.connection] - ValueError: Chunk too big
2022-12-30 18:30:47.772 [WARN ] [HABApp.openhab.connection] - Disconnected! Uncaught error in process_sse_events: Chunk too big
2022-12-30 18:31:20.017 [INFO ] [HABApp.openhab.connection] - Connected to OpenHAB version 3.4.0 (Release Build)
2022-12-30 18:31:21.506 [INFO ] [HABApp.openhab.items] - Updated 810 Items
2022-12-30 18:31:21.936 [INFO ] [HABApp.openhab.items] - Updated 20 Things
```

I have two servers, where one is my test machine, connected to the main machine with the remote openhab binding.
When import a new/modified item file to my main machine, I get these errors on my test machine.

**Comment by spacemanspiff2007** [comment]
> Here some examples of these status descriptions I've collected:

Could you post a a RestAPI response which shows one of those?
It's available through Developer Tools -> Api Explorer -> things -> `/things/{thingUID}`

**Comment by UrsusS** [comment]
I had answered by mail, but it did not come through...

Here is one example of a thing with status description.
```
{
  "channels": [],
  "statusInfo": {
    "status": "OFFLINE",
    "statusDetail": "COMMUNICATION_ERROR",
    "description": "Node is not communicating with controller"
  },
  "editable": true,
  "label": "Z-Wave Node 049",
  "bridgeUID": "zwave:serial_zstick:gehirn",
  "configuration": {
    "action_heal": false,
    "binding_pollperiod": 7200,
    "node_id": 49
  },
  "properties": {
    "zwave_class_basic": "BASIC_TYPE_ROUTING_SLAVE",
    "zwave_class_generic": "GENERIC_TYPE_METER",
    "zwave_frequent": "false",
    "zwave_neighbours": "",
    "zwave_listening": "true",
    "zwave_version": "0.0",
    "zwave_nodeid": "49",
    "zwave_lastheal": "2022-11-25T14:08:55Z",
    "zwave_routing": "true",
    "zwave_beaming": "true",
    "zwave_secure": "false",
    "zwave_class_specific": "SPECIFIC_TYPE_WHOLE_HOME_METER_SIMPLE"
  },
  "UID": "zwave:device:gehirn:node49",
  "thingTypeUID": "zwave:device"
}
```

**Comment by spacemanspiff2007** [comment]
Created https://github.com/openhab/openhab-core/issues/3309

**Comment by spacemanspiff2007** [comment]
@UrsusS 
Could you set the `HABApp.EventBus` logger to debug in the `logging.yml` and get the raw event from openHAB that contains the description information?

**Comment by UrsusS** [comment]
Here is one:
```
[2023-04-23 10:37:02,494] [  HABApp.EventBus.openhab]    DEBUG | {"topic":"openhab/things/fsinternetradio:radio:RadioWiGa/status","payload":"{\"status\":\"OFFLINE\",\"statusDetail\":\"COMMUNICATION_ERROR\",\"description\":\"java.util.concurrent.ExecutionException: java.net.NoRouteToHostException: Keine Route zum Zielrechner\"}","type":"ThingStatusInfoEvent"}
[2023-04-23 10:37:02,495] [          HABApp.EventBus]     INFO | fsinternetradio:radio:RadioWiGa: <ThingStatusInfoEvent name: fsinternetradio:radio:RadioWiGa, status: OFFLINE, detail: COMMUNICATION_ERROR>

```

**Comment by spacemanspiff2007** [comment]
@UrsusS Could you also provide a `ThingStatusInfoChangedEvent`?

**Comment by UrsusS** [comment]
Here it is:
```
[2023-05-19 05:22:52,797] [  HABApp.EventBus.openhab]    DEBUG | {"topic":"openhab/things/fsinternetradio:radio:TunerWoZi/statuschanged","payload":"[{\"status\":\"OFFLINE\",\"statusDetail\":\"COMMUNICATION_ERROR\",\"description\":\"java.util.concurrent.ExecutionException: java.net.NoRouteToHostException: Keine Route zum Zielrechner\"},{\"status\":\"OFFLINE\",\"statusDetail\":\"COMMUNICATION_ERROR\",\"description\":\"java.util.concurrent.TimeoutException: Total timeout 5000 ms elapsed\"}]","type":"ThingStatusInfoChangedEvent"}
[2023-05-19 05:22:52,797] [          HABApp.EventBus]     INFO | fsinternetradio:radio:TunerWoZi: <ThingStatusInfoChangedEvent name: fsinternetradio:radio:TunerWoZi, status: OFFLINE, detail: COMMUNICATION_ERROR, old_status: OFFLINE, old_detail: COMMUNICATION_ERROR>

```

**Comment by spacemanspiff2007** [comment]
Thanks for your help! The description will be part of the next release.
If you are eager you can try out the dev branch.

**Comment by UrsusS** [comment]
That's great, thanks a lot.
What is the recommended way to install the dev version? I used to use the offical branch.

**Comment by spacemanspiff2007** [comment]
[See docs](https://habapp.readthedocs.io/en/latest/installation.html#install-a-development-version-of-habapp). Be aware that there might be [breaking changes](https://github.com/spacemanspiff2007/HABApp/tree/Develop#changelog)

**Comment by spacemanspiff2007** [comment]
Fixed in 1.1.0

**Comment by UrsusS** [comment]
Thanks a lot, it works like a charm!
